README.md

```diff
@@ -77,13 +77,147 @@ on VA.gov, you'll need to build these static pages using the following commands:
 (e.g. `/preview?nodeId=XX`).
 
-If you do not have access to the SOCKS proxy, you can **fetch the latest cached version 
+If you do not have access to the SOCKS proxy, you can **fetch the latest cached version
 of the content** with the following:
 
 ```sh
 yarn fetch-drupal-cache
 ```
 
+#### How "templating" works
+"Templating" refers to the process of data representing website content being compiled into an HTML file.
+
+After set up, the `vagov-content` repo should be a sibling repo to `vets-website`. `vagov-content` contains static content files in the form of Markdown (`.md`) in the `vagov-content/pages` directory. At the top of each Markdown file, there are declarations of key-value pairs referred to as "frontmatter", which are often used to define metadata about the page or to describe how the page should be processed during the HTML compilation. The most common key in frontmatter is a `layout` property, which refers to a file in `vets-website/src/site/layouts`. For example, this [`layout`](https://github.com/department-of-veterans-affairs/vagov-content/blame/68a9ca3d87214bee2e14a95f54044cd5c7d19f5a/pages/coronavirus-chatbot.md#L3) property refers to this [layout file](https://github.com/department-of-veterans-affairs/vets-website/blob/6c61531163be7b52345b6096dc6a76fc38e7776e/src/site/layouts/page-breadcrumbs.html) in `vets-website`. The frontmatter of the `.md` file are passed to the layout file as the global scope.
+
+<details>
+
+<summary>Example of a Markdown processed throughout a layout file</summary>
+
+For example, pretend this is a `.md` in `vagov-content.`
+
+```markdown
+---
+layout: favorite-movies.html
+movies:
+  - Ace Ventura
+  - Disney Cars
+  - Toy Story
+---
+
+I love movies! Here are some of my recommendations.
+```
+
+The `layout` property would refer to a layout file `vets-website/src/site/layouts/favorite-movies.html`, which could look like -
+
+```
+<p>{{ contents }}</p>
+<ul>
+{% for movie in movies %}
+  <li>{{ movie }}</li>
+{% endfor %}
+</ul>
+```
+
+This templating language is called [Liquid](https://shopify.github.io/liquid/). The body of the Markdown file (which resides below the frontmatter of the page and may contain HTML) will be available to the template in a `contents` property. This would result in the following HTML snippet.
+
+```html
+<p>I love movies! Here are some of my recommendations.</p>
+<ul>
+  <li>Ace Ventura</li>
+  <li>Disney Cars</li>
+  <li>Toy Story</li>
+</ul>
+```
+
+</details>
+
+##### How templating works with CMS data
+
+After set up, your project should have issued a network request to fetch data from the VA CMS, Drupal. To verify, the response JSON should be visible in your local `vets-website/.cache/localhost/drupal` directory. To refresh the content, run `yarn build:content --pull-drupal`. The `vets-website/.cache` directory will be emptied and a new network request will be issued.
```

Comment:
We may want to add a note that you have to have your 🧦 on when running `--pull-drupal`